server/service/src/rnr_form/schedules_with_periods.rs

```diff
@@ -0,0 +1,63 @@
+use crate::service_provider::ServiceContext;
+
+use chrono::Utc;
+use repository::{
+    DateFilter, EqualFilter, Period, PeriodFilter, PeriodRepository, PeriodScheduleRow,
+    PeriodScheduleRowRepository, PeriodSort, PeriodSortField,
+    ProgramRequisitionSettingsRowRepository, RepositoryError,
+};
+
+#[derive(Debug)]
+pub struct PeriodSchedule {
+    pub schedule_row: PeriodScheduleRow,
+    pub periods: Vec<Period>,
+}
+
+pub fn get_schedules_with_periods_by_program(
+    ctx: &ServiceContext,
+    program_id: &str,
+) -> Result<Vec<PeriodSchedule>, RepositoryError> {
+    let settings_repo = ProgramRequisitionSettingsRowRepository::new(&ctx.connection);
+    let period_repo = PeriodRepository::new(&ctx.connection);
+
+    let settings = settings_repo.find_many_by_program_id(program_id)?;
```

Comment:
Maybe this just should have been another join in the period table?

Reply:
In most cases, I think it's fine to run multiple queries in a single request.

The more important goal is to make the code readable. In other projects I'd probably have leant towards a join here, but if we don't need data from the joined table, and with the extra complexity of diesel I think it's probably a good call to do it here.

There's a slight overhead to each query but should be minimal unless someone decides to run their database at the otherside of the world from their app server.

1vs1 queries isn't a big concern, n+1 problem is a bit more significant.
https://planetscale.com/blog/what-is-n-1-query-problem-and-how-to-solve-it
I think this is an N+1 thing as we need to run a query for every `period_schedule_id`

Assuming `N` is small ( e.g. there's not too many period schedules) this approach is fine.

Reply:
Its a good question! Probably only one active one? Will ask 🙏